src/steps/apply-transform.js

```diff
@@ -43,5 +47,11 @@ function transform(resource, req, res, transformMode, registry) {
     }
   };
 
-  return transformFn ? transformFn(resource, req, res, superFn) : resource;
+  if (!transformFn) {
+    return Promise.resolve(resource);
+  }
+
+  // Allow user to return a Promise or a value
+  let transformed = transformFn(resource, req, res, superFn);
+  return isPromise(transformed) ? transformed : Promise.resolve(transformed);
```

Comment:
This is redundant...if it's already a promise, `resolve` will keep it as is, so we can just return `Promise.resolve(transformed)`